src/lib/ITensorsExtensions/src/itensor.jl

```diff
@@ -58,18 +61,53 @@ function eigendecomp(A::ITensor, linds, rinds; ishermitian=false, kwargs...)
   D, U = eigen(A, linds, rinds; ishermitian, kwargs...)
   ul, ur = noncommonind(D, U), commonind(D, U)
   Ul = replaceinds(U, vcat(rinds, ur), vcat(linds, ul))
-
   return Ul, D, dag(U)
 end
 
-function map_eigvals(f::Function, A::ITensor, inds...; ishermitian=false, kwargs...)
+function map_eigvals(f::Function, A::ITensor, Linds, Rinds; kws...)
+  Linds = isa(Linds, Index) ? [Linds] : collect(Linds)
+  Rinds = isa(Rinds, Index) ? [Rinds] : collect(Rinds)
```

Comment:
Are there cases where we call this with `Linds::Index` and/or `Rinds::Index`? I would lean towards just not supporting that, and users have to call `map_eigvals(f, A, (i,), (j,))` or `map_eigvals(f, A, [i], [j])`.

Reply:
Yes I kind of would too... Especially if we are setting these functions up to be more like an "expert" back end to a future ITensor operator system. 

There were some tests that triggered this issue. Let me track them down and then the right move could be to just improve those tests.

Reply:
Ok, it was my own fault for writing a test with a line like 
```
sqrtM = map_eigvals(sqrt, M, s1', dag(s1); ishermitian=true)
```
which I've now changed to have brackets around the indices to make collections.

Do you have a feel for how we could get a better error message in that case? Like constraining the arguments or checking if the inputs satisfy the concept of a collection?

Reply:
I implemented this. Interestingly, after the check I caught some preexisting tests passing in individual indices. I guess that worked because it didn't interact with the newer `make_bosonic` code and the lower-down `eigen` call can handle that case. `eigen` does it by calling `indices` on its inputs. So we could go that direction too, as a way of "normalizing" the inputs and then allowing individual Index objects to be passed.